src/main/groovy/lang/ObjectRange.java

```diff
@@ -397,7 +392,10 @@ public boolean contains(Object value) {
 
     @Override
     public void step(int step, Closure closure) {
-        if (step == 0 && compareTo(from, to) == 0) {
+        if (compareEqual(step, 0)) {
+            if (compareNotEqual(from, to)) {
+                throw new GroovyRuntimeException("Infinite loop detected due to step size of 0");
```

Comment:
it was like this [in `IntRange`](https://github.com/paplorinc/incubator-groovy/blob/master/src/main/groovy/lang/IntRange.java#L360) also.
Would you like me to simplify it in both cases to no-ops?